### DISCUSSION TITLE
接入 Nextjs 应用作为子应用时 NextRouter 报错

### DISCUSSION THREAD

**Discussion by jianingroja** [web-infra-dev/modern.js#6498] [Q&A]
现有应用框架如下
- Next.js 主应用
  - Next.js 子应用
  - Next.js 子应用
主应用和子应用之间使用 Module Federation 提供的插件进行连接。

由于 Module Federation 宣布不再为 Next.js 16 及之后的版本提供插件支持，我正在尝试将主应用向 Modern.js 迁移。
期待达成的应用框架如下
- Modern.js 主应用
  - Next.js 主应用
  - Next.js 主应用

目前遇到的问题是，Next.js 子应用接入后，一旦涉及到子应用内部使用 `useRouter`，变产生报错，`NextRouter was not mounted`。

不知是否有开发者遇到相似问题。

**Comment by zllkjc** [comment]
这类问题我们更推荐提一个 issue 哈。目前我的猜测是，之前在子应用中使用的是主应用的 NextRouter，主应用变为 Modern.js 之后，外层不再有这个封装。

非常希望有一个 Demo 以便我们可以复现场景，并为这种迁移提供最佳实践~

**Reply by jianingroja** [reply]
多谢回复，已在这个新 [issue](https://github.com/web-infra-dev/modern.js/issues/6509) 中提出。
